lib/pure/ospaths.nim

```diff
@@ -157,64 +158,173 @@ const
     ## The character which separates the base filename from the extension;
     ## for example, the '.' in ``os.nim``.
 
+# TODO: MOVE to strutils, sequtils, or algorithms
+proc countWhile*(str: string, pred : proc(a:char):bool, prefix = true): int =
```

Comment:
You can use ``parseutils.parseWhile`` for this, or just ``str.len-str.replace("/", "").len``

If you want more complicated counting then just use a for loop...

Reply:
`parseWhile` is not as flexible (eg I need arbitrary lambda; and I need to go from both ends).

`str.len-str.replace("/", "").len` won't work with my other usages of `countWhile`.

> If you want more complicated counting then just use a for loop...

`countWhile` leads to simpler code